### ISSUE TITLE
CodeActions not appearing

### ISSUE THREAD

**Issue by nickgoddard777** [wbrakowski/AL-Navigator#80]
After the last vs code update, the Code Actions have stopped working.  I did change some settings and update some extensions.  I have tried though deleting all my extensions and reinstalling them and have also tried deleting all my settings but to no avail.

The Extensions I have installed now are the latest AL Language extension v8.1, And then the extensions from Waldo's Al Extension Pack.

I'm not sure if one of the other extensions is conflicting.  I'm still getting code actions from AL Toolbox so I don't think it can be a general Code action issue.

**Comment by wbrakowski** [comment]
Hey nickgoddard777,
thank you for reporting this. I had the same issues in the last days and thought it was only a local issue with my vs code. I will investigate this in the next days and post here when it is fixed.

**Comment by wbrakowski** [comment]
@nickgoddard777 Do you still have this issue? I checked it today and it looks like the problem disappeared without me having to fix anything.

**Comment by nickgoddard777** [comment]
Hi Waldemar,

First of all thanks for providing such an amazing extension.  I hadn't realised how much I use it until now, so thank you very much.

It still isn't working sorry.
  
I've got a copy of VS Code on my none Dev machine so I can play with the settings and extensions.  I deleted all settings and extensions and upgraded VS Code, which is now on v1.62.3.  I then installed the latest version of the AL Language Extension, which is on v8.2.545335.
I then installed your AL Navigator extension, which is on v0.4.6.  This then installed AZ AL Dev Tools/AL Code Outline (v3.0.22) and Waldo's CRS AL Language Extension (v1.4.11)
If I try and use the Sort Variables code action from AZ AL Dev Tools, this code action is there and works.
From your extension I tried CTRL-ALT-g and that took me to the Global Variables, and I tried CTRL-ALT-l, CTRL-ALT-a, CTRL-ALT-d and Shift-ALT-d and they work.  I then tried CTRL-ALT-k to go to keys but that didn't work.
If I enter a line such as CustomerNo := Customer."No."; The system will say it doesn't recognise CustomerNo so if I click into that and press CTRL+. I just get the pop-up that there is No Code Action Available.  If I leave the cursor on CustomerNo the Quick fix Lamp never appears.
I have then installed AL CodeActions as I was having issues with that extension as well but that now seems to be working.  I have noticed that David released a new version of this on Friday evening, and in his code, there is a commit saying 'Be compatible with AL version 8.2.545335'.  I don't really know about extension development, but in the code, he has changed parts about 'diagnostic'.  He has removed a line that says `switch (diagnostic.code) {` and replaced it with `switch (DiagnosticAnalyzer.getDiagnosticCode(diagnostic)) {`  I see in your [code ](https://github.com/wbrakowski/AL-Navigator/blob/2c98d631455346569a439d9d3015c8c93f8cb509/src/al/alCodeActionsProvider.ts) you have the same switch line.
In his code he has a new file for this diagnostic analyzer and in hat file there is this version check
```
if (semver.gte(microsoftExtension?.packageJSON.version, '8.2.545335'))
            return (d.code as {value: string, target: Uri}).value;
        else
            return d.code!.toString();
```

The URL for his commit is [here ](https://github.com/DavidFeldhoff/al-codeactions/commit/a3fc23b0df8cb65e8c52bc22d92db44b22c219dc)

I hope all that helps and if you need any help on anything please reach out.

![NoCodeActions](https://user-images.githubusercontent.com/332366/142852552-0a08ceb8-48c2-450c-8c35-b3b42f7b835d.png)

**Comment by nickgoddard777** [comment]
I've just downgraded my version of AL Language Extension to v8.1.54.0594 and it now works.  I have noticed while I was typing my last comment you have released a new version, that doesn't seem to have fixed this issue.  I guess MS must have changed something in v8.2.545335 that has broken your extension.

**Comment by wbrakowski** [comment]
@nickgoddard777 It should be fixed with the new version 0.4.8. I will close this. Feel free to reopen if you notice the problem again. Thanks again and have a nice weekend.
